### PR TITLE
Add Total Amount Feature to Donations Page

### PR DESCRIPTION
**Pull request by ertugrul-aydin**

### CODE REVIEW

File: apps/dashboard/pages/donations.tsx [hunk 1]
```diff
@@ -81,7 +86,15 @@ const DonationsPage = () => {
         currentPage={currentPage}
         setCurrentPage={setCurrentPage}
         onSort={setSort}
-      />
+      >
```
Comment (7alip) [inline apps/dashboard/pages/donations.tsx:89]: You don't need to show the children if `totalAmount` doesn't exist

File: apps/dashboard/pages/donations.tsx [hunk 2]
```diff
@@ -81,7 +86,15 @@ const DonationsPage = () => {
...
+        <Spacer />
+        <Flex textAlign={'right'}>
+          <Spacer />
+          <Box paddingY={2} paddingX={5} bg="white" shadow="base">
+            Total Amount: <Text as="b">{totalAmount}</Text>
+          </Box>
+        </Flex>
```
Comment (7alip) [inline apps/dashboard/pages/donations.tsx:96]: Having `Spacer`s along with `textAlight=right` seems overkill to me.

Can something like this work:

```tsx
<Flex justify="end">
  <Box />
</Flex>
```

**Review by 7alip — COMMENTED** [review]
Thanks

File: packages/ui/src/admin/DataTables/DataTable.tsx [hunk 3]
```diff
@@ -23,6 +24,7 @@ export const DataTable = <T extends StrapiModel>({
           <Box>{t('no-data')}</Box>
         )}
       </Box>
+      {tableProps.data.length > 0 && children}
```
Comment (7alip) [inline packages/ui/src/admin/DataTables/DataTable.tsx:27]: What if we want to display something as DataTable children even though there is no data?

File: apps/dashboard/pages/donations.tsx (hunk 1 above)
Comment (7alip) [reply apps/dashboard/pages/donations.tsx:89]: I would control displaying totalAmount here instead of `{tableProps.data.length > 0 && children}`.

Comment (ertugrul-aydin) [reply apps/dashboard/pages/donations.tsx:89]: Yes, you are right. i will make the change.

File: packages/ui/src/admin/DataTables/DataTable.tsx (hunk 3 above)
Comment (ertugrul-aydin) [reply packages/ui/src/admin/DataTables/DataTable.tsx:27]: Yes, you are right. i will make the change.

File: apps/dashboard/pages/donations.tsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { useEffect, useState } from 'react'
 
-import { useUpdateEffect } from '@chakra-ui/react'
+import { useUpdateEffect, Box, Spacer, Text, Flex } from '@chakra-ui/react'
```
Comment (7alip) [inline apps/dashboard/pages/donations.tsx:3]: Lint error, Spacer is defined but never used

```suggestion
import { useUpdateEffect, Box, Text, Flex } from '@chakra-ui/react'
```

**Review by 7alip — APPROVED** [review]
Thanks

**Comment by 7alip** [comment]
> If I see correct, amount show paid and unpaid amount. If it true, It should be changed. We need only paid amount.

@MustafaTRKYLMZ Shouldn't we adjust it with status filter?  I don't know if it makes sense but we can still see what is the total amount of unpaid transactions.

**Comment by ertugrul-aydin** [comment]
> If I see correct, amount show paid and unpaid amount. If it true, It should be changed. We need only paid amount.

I've updated it to display both 'paid' and 'unpaid' amounts separately. What do you think about this ? I have shared screenshot below.

Ekran Resmi 2023-09-18 12 32 46

**Comment by 7alip** [comment]
> I've updated it to display both 'paid' and 'unpaid' amounts separately. What do you think about this ? I have shared screenshot below.

It's also fine for me, thanks

**Comment by MustafaTRKYLMZ** [comment]
> > If I see correct, amount show paid and unpaid amount. If it true, It should be changed. We need only paid amount.
> 
> @MustafaTRKYLMZ Shouldn't we adjust it with status filter? I don't know if it makes sense but we can still see what is the total amount of unpaid transactions.

In mijn opinion We don't need unpaid amount. I think paid amount is enought. For example one person can try lots of time.

File: apps/dashboard/pages/donations.tsx [hunk 5]
```diff
@@ -64,6 +64,21 @@ const DonationsPage = () => {
...
+        return acc + (donation.amount || 0)
+      }, 0)
+
+  const totalUnpaid =
+    donations &&
+    donations
+      .filter(donation => donation.status !== 'paid')
+      .reduce((acc, donation) => {
+        return acc + (donation.amount || 0)
+      }, 0)
```
Comment (7alip) [inline apps/dashboard/pages/donations.tsx:81]: Our conclusion is to list only paid transactions. So we don't need this part

File: apps/dashboard/pages/donations.tsx [hunk 6]
```diff
@@ -81,7 +96,17 @@ const DonationsPage = () => {
...
         setCurrentPage={setCurrentPage}
         onSort={setSort}
-      />
+      >
+        {donations && (
+          <Flex justify={'end'}>
+            <Box paddingY={2} paddingX={5} bg="white" shadow="base">
+              Total Paid: <Text as="b">{totalPaid}</Text>
+              <br />
+              Total Unpaid: <Text as="b">{totalUnpaid}</Text>
```
Comment (7alip) [inline apps/dashboard/pages/donations.tsx:105]: Remove unpaid part

File: apps/dashboard/pages/donations.tsx [hunk 7]
```diff
@@ -64,6 +64,21 @@ const DonationsPage = () => {
 
   const donations = donationsQuery?.data?.data as Donation[]
   const totalCount = donationsQuery?.data?.meta?.pagination?.pageCount || 0
+  const totalPaid =
```
Comment (7alip) [inline apps/dashboard/pages/donations.tsx:67]: Update donationsQuery filter, only fetch them based on paid status.

`totalPaid` can be renamed to `totalAmount`

**Review by 7alip — CHANGES_REQUESTED** [review]
We decided to display always only paid donations. Can you make the changes?

**Review by 7alip — COMMENTED** [review]
Thank you,

There are some conflicts. Would you prefer to resolve them yourself?

**Pull request merged**